### ISSUE TITLE
Wildcard cert with non publicly signed root CA for notary

### ISSUE THREAD

**Issue by svshvets** [goharbor/harbor#13023]
Hello experts,

I am working with harbor, v. 2.0.2-e91b4ff1 on ubuntu v. 20.04. I have followed the following guide to install and configure https://goharbor.io/docs/2.0.0/install-config/run-installer-script. I have configured notary-server and notary-signer on the same ubuntu host. I have a question about certs for notary-server and notary-signer, specifically on whether it is required to use publicly signed CA root in order for notary to be able to sign images. I currently have internal certs for notary-server and signer. But the reason I'm asking about public CA root is because I'm receiving the following error when trying to push and sign an image: "Error: error contacting notary server: x509: certificate signed by unknown authority" 
![image](https://user-images.githubusercontent.com/71026805/92625330-1fbd5100-f28e-11ea-9343-0ca760c80173.png)

The reason I'm asking about public CA root is also because I initially had an issue with docker daemon where I was trying to use internal wildcard cert and was getting similar error but once I've used an external cert the issue got resolved. 
I've also added CA root of the current internal cert into ca-bundles on ubuntu server but that doesn't seem to resolve the error I am seeing.  Am I missing something or do I in fact have to use public CA root, wanted to make sure internal certs can't be used before I went purchasing. 
I'm also getting trust error when attempting to add a signer to a project as well which I believe is related to same above error 
![image](https://user-images.githubusercontent.com/71026805/92625141-ebe22b80-f28d-11ea-9032-ef564cde9181.png)

Also to add, when I login to notary server, I get same error:

![image](https://user-images.githubusercontent.com/71026805/92643452-6c615600-f2a7-11ea-85e7-d8a7be44ef42.png)

Thank you for any help!

**Comment by reasonerjt** [comment]
hello, 

As for the error in the first screenshot.

this certificate validation happens when the docker client communicate with the "DOCKER_CONTENT_TRUST_SERVER"

you need to copy the ca cert to `~/.docker/tls/$DOCKER_CONTENT_TRUST_SERVER`

**Comment by svshvets** [comment]
reasonerjt,

I was able to resolve the errors and it appears the issue was with using non publicly signed root CA, I had cert in the location that you have mentioned above. Right now I have certs matching between harbor and notary-server with publicly signed root and the error message for certs has gone away, however I'm now getting a different error message. When I login to harbor, for example 'docker login harbor-server.domain,com' I successfully authenticate (I can either authenticate using internal admin account or any domain account) but if I attempt to do the same to notary-server, for example 'docker login notary-server.domain.com:4443' I get the following error message: 
![image](https://user-images.githubusercontent.com/71026805/93343300-f48ab280-f7f5-11ea-910e-998cf6248ef2.png)
 
I get same error of Unauthorized when I try to push an image now as well. Now, if I do remove authentication part from server-config.postgres.json config file then pushing an image works, however when I login to harbor I still see an image as being unsigned, so that tells me that authentication piece is required in order for signing to work but I haven't figured out what I need to have here in order to allow ldap authentication that I have configured for harbor:  
![image](https://user-images.githubusercontent.com/71026805/93343835-95796d80-f7f6-11ea-8d85-49920e7402be.png)

I feel like I'm close to getting it to work but authentication piece is the last part to get working, any suggestions are greatly appreciated. I've been going over this guide for notary server config file, but still not sure what I need to specify in order to allow LDAP auth that is configured for harbor - https://docs.docker.com/notary/reference/server-config/

Thanks
